polaris_wyre/wyre/integration.py

```diff
@@ -0,0 +1,147 @@
+        of the payment transaction to ``Transaction.receiving_anchor_account``,
+        the string representation of the memo the client should attach to the
+        transaction to ``Transaction.memo``, and the type of that memo to
+        ``Transaction.memo_type``.
+        This method is only called once for a given transaction. The values
+        saved will be returned to the client in the response to this request or
+        in future ``GET /transaction`` responses.
+        **Polaris assumes ``Transaction.save()`` is called within this method.**
+        The memo saved to the transaction object _must_ be unique to the
+        transaction, since the anchor is expected to match the database record
+        represented by `transaction` with the on-chain transaction submitted.
+        This function differs from ``get_distribution_account()`` by allowing the
+        anchor to return any Stellar account that can be used to receive a payment.
+        This is ideal when the account used is determined by a custody service
+        provider that does not guarantee that the account provided will be the
+        account provided for future transactions.
+        :param request: the request that initiated the call to this function
+        :param transaction: the transaction a Stellar account and memo must be
+            saved to
+        """
+        stellar_account_address, user_id = self.wyre.get_account()
+        transaction.receiving_anchor_account = stellar_account_address
+        # TODO: Find a way to let the memo unique when integrating with Wyre
+        transaction.memo_type = Transaction.MEMO_TYPES.text
+        transaction.memo = user_id
+        transaction.save()
+
+    def submit_deposit_transaction(
+        self, transaction: Transaction, has_trustline: bool = True
+    ) -> dict:
+        """
+        Submit the transaction to the Stellar network using the anchor's custody
+        service provider and return the JSON body of the associated
+        ``GET /transaction/:id`` Horizon response.
+        If ``self.claimable_balances_supported`` is ``True``, Polaris may call
+        this method when the destination account does not yet have a trustline
+        to ``Transaction.asset``. In this case, the anchor should send the
+        deposit as a claimable balance instead of a payment or path payment. Use
+        the `has_trustline` parameter to determine which operations to use.
+        If ``self.claimable_balances_supported`` is ``False``, this method will only
+        be called when the destination account exists and has a trustline to
+        ``Transaction.asset``.
+        :param transaction: the ``Transaction`` object representing the Stellar
+            transaction that should be submitted to the network
+        :param has_trustline: whether or not the destination account has a trustline
+            for the requested asset
+        """
+        amount = round(
```

Comment:
As a curiosity, how was this rounding method decided? Were all other [roudings](https://realpython.com/python-rounding/) considered?

Reply:
That's a pattern from `Polaris` since you can define how many decimals places you will consider when doing the transaction. You set that on the `Asset` model on field `significant_decimals`. You can look [here](https://github.com/stellar/django-polaris/blob/f53a20ca4381502374ce6e5ab2d7c097f08961ee/polaris/polaris/utils.py#L366-L369).